src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneState = Object.assign({}, state);
+  const arrayOfStates = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          delete cloneState[property];
+        };
+
+        break;
+
+      case 'clear':
+        Object.keys(cloneState).forEach(key => delete cloneState[key]);
+        break;
+
+      default:
+        break;
+    }
```

Comment:
I think, you should add some act to default. `return 'Error';` or `throw new Error('actions is not correct');`